design/ceph/object/store.md

```diff
@@ -323,31 +323,113 @@ spec:
+
+This is different from CephObjectZone `customEndpoints` which is used for configuring the object store to replicate and sync data amongst Multisite.
+
+When endpoints are configured in `rgw_dns_names`, the RGW will reject any incoming connections
+intended for endpoints not in the list. Therefore, all endpoints that might be used must be added.
+
+For convenience, and to ensure other CephObjectStore configurations are not rendered unusable when
+users add additional endpoints to `dnsNames`, the following should be added to the `rgw_dns_names`
+list automatically by Rook:
+- the `advertiseEndpoint.dnsName`
+- the default service endpoint for the object store (e.g., `rook-ceph-rgw-my-store.rook-ceph.svc`)
+- CephObjectZone `customEndpoints`
+
+For a `hosting` configuration that includes only `dnsNames`, there is potential internal/developer
+and user confusion when adding more than one entry to the `dnsNames` list. From the user
+perspective, Rook might make undesirable assumptions about which endpoint to advertise for
+CephObjectStoreUser, OBC, or COSI resources. From the internal/developer perspective, it might be
+impossible to make a well-informed assumption in every scenario. Therefore, add a required
+`advertiseEndpoint` configuration which ensures both users and Rook have clear expectations about
+behavior. `dnsNames` can then be used for optional, additional endpoints since `advertiseEndpoint`
+will be certain to contain at a single `rgw_dns_names` entry.
+
+If users wish to add an external ingress to the `dnsNames` list but still use Rook's default
+CephObjectStore service internally, they can add the service endpoint to the `advertiseEndpoint`,
+which ensures clarity for users about Rook behavior.
+
+S3 clients (and therefore OBCs and COSI) generally assume a single endpoint for an object store.
+Therefore, Rook supports only a single `advertiseEndpoint`. Multiple will not be supported to avoid
+user and internal/developer confusion.
+
+When this feature is enabled, there is also ambiguity about which endpoint Rook should use for Admin
+Ops API communication. Some users have reported issues with Rook using a `dnsNames` endpoint
+(or `advertiseEndpoint`) when they wish to set up ingress certificates after Rook deployment. The
+obvious alternative is to have Rook always use the CephObjectStore service, but other users have
+expressed troubles creating certificates or CAs that allow the service endpoint in the past.
+
+In order to attempt to strike the best balance for everyone, and to provide the best clarity for
+users and Rook internally, Rook will always use the service endpoint for admin ops. Rook
+documentation must inform users that if TLS is enabled, they must give Rook a certificate that
+accepts the service endpoint. Alternately, if that is not possible, Rook will add an
+`insecureSkipTlsVerification` option to the CephObjectStore to allow users to provision a healthy
```

Comment:
Agreed, makes sense to expose this option, then we can have different defaults. For example, in `object-test.yaml` it would be `true`, while in the production example `object.yaml`, it would be false.

Reply:
For comparison, the other option I have considered is that we could add a field `useForAdminOperations: true|false` to the `advertiseEndpoint` config. 

When I think about that from the user view, it seems like it wouldn't be as broadly useful as the `insecureSkipTlsVerify` option, and it might add complexity that exposes internal details without need. My plan is to keep this option in mind for the future in case we get user feedback about continued challenges with TLS certs on the Service but not to plan for it right now.